src/App.tsx

```diff
@@ -1,61 +1,126 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import React, { useState } from 'react';
+
+import { TodoList } from './components/TodoList';
+import { User } from './components/UserInfo';
+
+export type Todo = {
+  id: number;
+  title: string;
+  userId: number;
+  completed: boolean;
+  user: User | null;
+};
+
+function getUserById(userId: number) {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+export const todos = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+function getTodoId(initialTodos: Todo[]) {
+  const todoArrId = initialTodos.map(todo => todo.id);
+
+  return Math.max(...todoArrId) + 1;
+}
```

Comment:
[CODE KNOWLEDGE] - The function `getTodoId` is used to generate a new ID for a todo by finding the highest current ID and adding one. While this works, it's not the most reliable way to generate unique IDs, especially in a collaborative or persistent environment. Consider using a library like `uuid` for generating unique IDs that don't rely on the current state of the todos array.